bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/partitiontable/dialog/ComboBoxTextCellEditor.java

```diff
@@ -0,0 +1,37 @@
+package com.espressif.idf.ui.partitiontable.dialog;
+
+import org.eclipse.jface.viewers.ComboBoxCellEditor;
+import org.eclipse.swt.custom.CCombo;
+import org.eclipse.swt.widgets.Table;
+
+public class ComboBoxTextCellEditor extends ComboBoxCellEditor
+{
+	public ComboBoxTextCellEditor(Table table, String[] properties, int style)
+	{
+		super(table, properties, style);
+	}
+
+	@Override
+	protected void doSetValue(Object value)
+	{
+		if (value instanceof String)
+		{
+			((CCombo) getControl()).setText((String) value);
+		}
+		else
+		{
+			super.doSetValue(value);
+		}
```

Comment:
are we sure that we can have a value which is not instance of String if not we can probably log an exception here

Reply:
Hi, @alirana01. Thanks for the review. In this combo box, we can receive String or int values. This is because this field is editable and we can enter a value that is not in the combo box